File: scala-spring/eventstore-examples/src/main/scala/net/chrisrichardson/eventstore/examples/bank/accounts/AccountService.scala
package net.chrisrichardson.eventstore.examples.bank.accounts

import net.chrisrichardson.eventstore.EventStore
import net.chrisrichardson.eventstore.examples.bank.accounts.AccountCommands.OpenAccountCommand
import net.chrisrichardson.eventstore.util.ServiceUtil._

class AccountService(implicit eventStore : EventStore) {

  def openAccount(initialBalance : BigDecimal) =
    newEntity[Account] <== OpenAccountCommand(initialBalance)

}

File: java-spring/accounts-command-side-backend/src/main/java/net/chrisrichardson/eventstore/javaexamples/banking/backend/commandside/accounts/AccountService.java
package net.chrisrichardson.eventstore.javaexamples.banking.backend.commandside.accounts;



import io.eventuate.AggregateRepository;
import io.eventuate.EntityWithIdAndVersion;

import java.math.BigDecimal;
import java.util.concurrent.CompletableFuture;

public class AccountService  {

  private final AggregateRepository<Account, AccountCommand> accountRepository;

  public AccountService(AggregateRepository<Account, AccountCommand> accountRepository) {
    this.accountRepository = accountRepository;
  }

  public CompletableFuture<EntityWithIdAndVersion<Account>> openAccount(BigDecimal initialBalance) {
    return accountRepository.save(new OpenAccountCommand(initialBalance));
  }

}
